tensorrt_llm/_torch/attention_backend/fmha/msa_sparse_gqa.py

```
@@ -0,0 +1,167 @@
+        # intermittently corrupt the flashinfer dense-attention path.
+        # See MsaProxyMqaFmha.is_available for the full story; the real
+        # import happens at first kernel use in forward_block_sparse.
+        if importlib.util.find_spec("fmha_sm100") is None:
+            logger.debug("MsaSparseGqaFmha is unavailable: fmha_sm100 package not installed.")
+            return False
+        if not torch.cuda.is_available():
+            logger.debug("MsaSparseGqaFmha is unavailable: no CUDA device.")
+            return False
+        try:
+            major, _ = torch.cuda.get_device_capability()
+        except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)